### PR TITLE
`podman-compose` support

### PR DESCRIPTION
**Pull request by roberth**

### CODE REVIEW

**Comment by roberth** [comment]
We don't really need podman-compose as of now, because we support podman via docker-compose. The `json.type` change is good though.

**Comment by DieracDelta** [comment]
Is it possible to tell arion to build with podman without the workarounds proposed [here](https://github.com/hercules-ci/arion/issues/122)? Trying to figure out how to deploy with Arion and podman on non-nixos systems.

**Comment by roberth** [comment]
@DieracDelta It seems that rootless podman is becoming a [viable](https://github.com/hercules-ci/arion/issues/125#issuecomment-906945113) option, but it needs to be wired into the arion command to be easy to use. I am not sure if we need to worry about having a docker socket, but regardless, it seems quite feasible.

Nice handle btw. I suppose you like to neutralize convoluted functions?

**Comment by DieracDelta** [comment]
Hah yes the handle was an (un)fortunate side effect of my signal processing class. Lots of fun but also lots of scary math 

Yeah, it seems easy enough for any Linux user to get Arion working with podman. We're still looking into how easy this is to spin up on Mac and Windows. But good to know we're not missing anything in the arion cli. (This is an amazing tool BTW. Surprised there are so few usages on github)

**Comment by roberth** [comment]
Podman can serve as a backend for docker compose nowadays, so I don't think anyone needs podman compose anymore.
Maybe I'm wrong, but until I learn more, long live podman `docker compose` :tada:

**Comment by Enzime** [comment]
To run rootless Podman with Arion, I ended up having to alias `podman-compose` to `docker-compose`

https://github.com/Enzime/dotfiles-nix/blob/%F0%9F%94%A5/overlays/docker-compose.nix#L1-L8

Would love to see some simple way to call `podman-compose` instead of `docker-compose` from Arion :)

File: src/nix/modules/composition/docker-compose.nix [hunk 1]
```diff
@@ -29,6 +29,20 @@ let
       config.service.name = name;
     };
 
+  json.type = with lib.types; let
```
Comment (roberth) [inline src/nix/modules/composition/docker-compose.nix:32]: we can use `(pkgs.formats.json {}).json` by now.

**Comment by roberth** [comment]
> rootless

Can rootless podman provide a docker socket? That would give us the best of both worlds and keep the number of code paths _significantly_ lower. It will be similar to rootful podman with `docker-compose`, currently documented as the recommended [installation method for NixOS](https://docs.hercules-ci.com/arion/#_nixos).



Otherwise, I suppose this PR could be revived, but I don't have the bandwidth to do it.

What's missing for this PR to be merged seems to be
 - see which existing tests are compatible and run them in ci
 - documentation for how to use it and what the (significant?) limitations are

**Pull request closed without merging**